amen/time.py

```diff
@@ -17,7 +20,40 @@ def __init__(self, time, duration, audio, unit='s'):
 
     def __repr__(self):
         args = self.time.delta * 1e-9, self.duration.delta * 1e-9
-        return '<TimeSlice, start: {0:.2f}, duration: {1:.2f}'.format(*args)
+        return '<TimeSlice, start: {0:.2f}, duration: {1:.2f}>'.format(*args)
+
+    def get_samples(self):
+        """
+        Gets the samples corresponding to this TimeSlice from the parent audio object.
+        """
+        start = self.time.delta * 1e-9
+        duration = self.duration.delta * 1e-9
+        starting_sample, ending_sample = librosa.time_to_samples([start, start + duration], self.audio.sample_rate)
+
+        left_offsets = self._get_offsets(self.audio.raw_samples[0], starting_sample, ending_sample)
+        right_offsets = self._get_offsets(self.audio.raw_samples[1], starting_sample, ending_sample)
+
+        samples = self._offset_samples(starting_sample, ending_sample, left_offsets, right_offsets)
+
+        return samples, left_offsets, right_offsets
+
+    def _get_offsets(self, channel, starting_sample, ending_sample):
```

Comment:
I think that this should just do:

`left_offsets, right_offsets = self._get_offsets(starting_sample, ending_sample)`